src/dotnet/commands/dotnet-tool/install/ProjectRestorer.cs

```diff
@@ -44,7 +44,7 @@ public ProjectRestorer(IReporter reporter = null)
             {
                 "--runtime",
                 AnyRid,
-                $"-property:BaseIntermediateOutputPath={assetJsonOutput.ToXmlEncodeString()}"
+                $"-property:BaseIntermediateOutputPath={assetJsonOutput.Value}"
```

Comment:
Does this work for directories that contain special MSBuild characters, such as `;`?

Reply:
no, same as `,` But so is `dotnet build -o `. I think to be on the same level of `dotnet build` should be enough for this 2.1.3xx fix

Reply:
My conclusion is there is not standard way to do escape. Maybe escape every control char (> , ;) with url data encoding but keep non ascii should be the right way (but it is so odd). That's why i hesitate to put it in 2.1.3xx

Reply:
At any rate, I think we should have double quotes around the property's value to ensure MSBuild interprets the contents as a literal, even if semicolon won't work in the `BaseIntermediateOutputPath` property.

Reply:
https://github.com/dotnet/cli/blob/cc80822c16b955a1f664f8a65ed2c76c3c52a824/src/Microsoft.DotNet.Cli.Utils/CommandResolution/GenericPlatformCommandSpecFactory.cs#L17

It is added by CommandSpecFactory layer

Reply:
And -o is doing the same https://github.com/dotnet/cli/blob/6f6f28dac6936352049cad0a03a3a06b8f8ceb88/src/dotnet/commands/dotnet-build/BuildCommandParser.cs#L27

Reply:
`ArgumentEscaper.EscapeAndConcatenateArgArrayForProcessStart` quotes the entire argument string as-needed, which is not the same thing as quoting the property value.  `ArgumentEscaper` will **escape** the double quotes so that MSBuild can parse them on its command line.  When doing so, it will interpret the value **literally** rather than seeing a comma or semicolon and treating it as a multi-property option (e.g. `/p:First=foo,Second=bar`). 

You can check for yourself the semantic difference if you pass `--tool-path foo,bar` to `dotnet tool install`.  Without the double quotes, it fails.  With the double quotes, it succeeds.

The `-o` option for `dotnet build` is a bug.  I shouldn't have to quote it on the command line like `-o '"foo,bar"'` (bash/zsh) or `-o \"foo,bar\"` (Windows) for it to respect the contents literally.

Reply:
We already know dotnet build -o cannot handle ';'. There is a bug in how we handle msbuild params for sure. However my stand for 2.1.3xx is to keep the same as -o (won't be worse). Or we should all in to complete understand what is the right thing to do and do the change. Which could delay the fix for at least a week. At the same time, if you pass a quoted "c:\temp" in xml and pass it to BaseIntermediateOutputPath, it will try to interpret it as CURRENTDIRECTOY\"c:\temp" for some reason.

Reply:
We shouldn't copy a bug just for parity.  Use double quotes.  Note how my examples above all used comma, which should work with both `-o` (it doesn't because it's a bug) and `--tool-path` (let's not copy that bug, ok?) without the user having to run through any hoops.

Reply:
As long as you are sure it will not introduce more bugs. After discover CURRENTDIRECTOY"c:\temp"  case at least I am not so sure

Reply:
I'm not sure what you mean by "quoted in XML".  In the project file?  It should never be quoted there because they would be literal quotes.

Passed on the command line, the quotes will be respected and removed during MSBuild's command line parsing.

Reply:
I talked to Peter offline. We agree on this version. Since there is no solution can solve all the cases.

And I talked to @livarcocc later. He think this solution is also good. It supports a upper set of characters. We should keep investigate a better solution for 2.1.4xx.

Reply:
👍